sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py

```diff
@@ -4,39 +4,48 @@
 # license information.
 # --------------------------------------------------------------------------
 import logging
-from typing import NoReturn, TYPE_CHECKING
+from typing import (
+    Any, Collection, Dict, List, NoReturn, Tuple, Union,
+    TYPE_CHECKING
+)
 from xml.etree.ElementTree import Element
 
-from azure.core.pipeline.policies import ContentDecodePolicy
 from azure.core.exceptions import (
-    HttpResponseError,
+    ClientAuthenticationError,
     DecodeError,
+    HttpResponseError,
+    ResourceExistsError,
     ResourceModifiedError,
-    ClientAuthenticationError,
     ResourceNotFoundError,
-    ResourceExistsError
 )
+from azure.core.pipeline.policies import ContentDecodePolicy
 from ._models import (
-    FileProperties,
+    AnalyticsLogging,
+    DeletedPathProperties,
     DirectoryProperties,
+    FileProperties,
     LeaseProperties,
-    DeletedPathProperties,
-    StaticWebsite,
-    RetentionPolicy,
     Metrics,
-    AnalyticsLogging,
-    PathProperties
+    PathProperties,
+    RetentionPolicy,
+    StaticWebsite
 )
 from ._shared.models import StorageErrorCode
 from ._shared.response_handlers import deserialize_metadata
 
 if TYPE_CHECKING:
-    pass
+    from azure.core.rest import RestRequestsTransportResponse as PipelineResponse
```

Comment:
The `response`s in datalake deserialize are not technically `PipelineResponse`. Stepping thru the code, it seems like the `response` is actually `RestRequestsTransportResponse` from `azure.core.rest`. However, this type is not exported. While this makes the `mypy` errors disappear (except for one that states this type is not exported), it might not fly for tests, etc.

Reply:
`azure/storage/filedatalake/_deserialize.py:37: error: Module "azure.core.rest" has no attribute "RestRequestsTransportResponse"  [attr-defined]`